util/host-utils.c

```diff
@@ -266,3 +266,154 @@ void ulshift(uint64_t *plow, uint64_t *phigh, int32_t shift, bool *overflow)
         *plow = *plow << shift;
     }
 }
+/*
+ * Unsigned 256-by-128 division.
+ * Returns the remainder via r.
+ * Returns lower 128 bit of quotient.
+ */
+static Int128 udiv256_qrnnd(Int128 *r, Int128 n0, Int128 n1, Int128 d)
```

Comment:
Any good reason to switch n0 and n1, in comparison to udiv_qrnnd? This will definitely cause confusion.